src/MonoMod.RuntimeDetour/DetourManager.Managed.cs

```diff
@@ -360,6 +360,10 @@ internal sealed class ManagedDetourState {
 
                     UpdateEndOfChain();
                     UpdateChain(ilhook.Factory, out _);
+                } catch {
+                    // an exception ocurred while trying to install the ILHook, we should revert
+                    RemoveILHook(ilhook, takeLock: false);
```

Comment:
This will call `InvokeDetourEvent(DetourManager.DetourUndone, DetourUndone, detour);` which seems like poor form. 

Otherwise looks good